tests/parametric/test_telemetry.py

```diff
@@ -164,6 +164,144 @@ def test_library_settings(self, library_env, test_agent, test_library):
+                "DD_LOG_LEVEL": "error",
+                "OTEL_LOG_LEVEL": "debug",
+                "DD_TRACE_SAMPLE_RATE": "0.5",
+                "OTEL_TRACES_SAMPLER": "traceidratio",
+                "OTEL_TRACES_SAMPLER_ARG": "0.1",
+                "DD_TRACE_ENABLED": "true",
+                "OTEL_TRACES_EXPORTER": "none",
+                "DD_RUNTIME_METRICS_ENABLED": "true",
+                "OTEL_METRICS_EXPORTER": "none",
+                "DD_TAGS": "foo:bar,baz:qux",
+                "OTEL_RESOURCE_ATTRIBUTES": "foo=bar1,baz=qux1",
+                "DD_TRACE_PROPAGATION_STYLE": "datadog",
+                "OTEL_PROPAGATORS": "datadog,tracecontext",
+                "OTEL_LOGS_EXPORTER": "none",
+                "OTEL_SDK_DISABLED": "false",
+            }
+        ],
+    )
+    def test_telemetry_otel_env_hiding(self, library_env, test_agent, test_library):
+        with test_library.start_span("test"):
+            pass
+        event = test_agent.wait_for_telemetry_event("generate-metrics", wait_loops=400)
+        payload = event["payload"]
+        assert event["request_type"] == "generate-metrics"
+
+        metrics = payload["series"]
+
+        assert payload["namespace"] == "tracers"
+
+        otelHiding = [s for s in metrics if s["metric"] == "otel.env.hiding"]
+        otelInvalid = [s for s in metrics if s["metric"] == "otel.env.invalid"]
+
+        assert len(otelHiding) == 9
+        assert len(otelInvalid) == 0
+
+        expected_tags = [
+            ["DD_TRACE_LOG_LEVEL", "OTEL_LOG_LEVEL"]
+            if context.library == "nodejs"
+            else ["DD_LOG_LEVEL", "OTEL_LOG_LEVEL"],
+            ["DD_TRACE_PROPAGATION_STYLE", "OTEL_PROPAGATORS"],
+            ["DD_SERVICE", "OTEL_SERVICE_NAME"],
+            ["DD_TRACE_SAMPLE_RATE", "OTEL_TRACES_SAMPLER"],
+            ["DD_TRACE_SAMPLE_RATE", "OTEL_TRACES_SAMPLER_ARG"],
+            ["DD_TRACE_ENABLED", "OTEL_TRACES_EXPORTER"],
+            ["DD_RUNTIME_METRICS_ENABLED", "OTEL_METRICS_EXPORTER"],
+            ["DD_TAGS", "OTEL_RESOURCE_ATTRIBUTES"],
+            ["DD_TRACE_OTEL_ENABLED", "OTEL_SDK_DISABLED"],
+        ]
+
+        for expected in expected_tags:
+            assert any(all(tag in metric["tags"] for tag in expected) for metric in otelHiding)
+
+        for metric in otelHiding:
+            assert metric["points"][0][1] == 1
+
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    @missing_feature(context.library == "cpp", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
```

Comment:
we are not running this test for any language?

Reply:
After an update to the RFC I had to make changes to the Node.js implementation. I have a PR up for that reimplementation in this PR https://github.com/DataDog/dd-trace-js/pull/4369 which I have tested against locally with those system tests